assets/app/view/game/round/stock.rb

```diff
@@ -350,6 +354,16 @@ def render_buy_input_interval(company)
             *buy_buttons,
           ])]
         end
+
+        def render_bid_input(company)
+          return [] if @step.max_bid(@current_entity, company) < @step.min_bid(company)
+          return [] if @current_actions.include?('bidding_tokens') &&
+            @step.highest_player_bid?(@current_entity, company)
+          return [] if @current_actions.include?('bidding_tokens') && @step.find_bid(@current_entity, company).nil? &&
+            @step.bidding_tokens(@current_entity).zero?
```

Comment:
The UI shouldn't have to make these logic decisions. Can we replace with `@step.can_bid?(@current_entity, company)`?

Reply:
Yep that i can do.